front/lib/api/files/tool_output.ts

```diff
@@ -68,3 +72,97 @@ export async function internalCreateToolOutputFile(
 
   return file;
 }
+
+export async function internalCreateSearchableTextFile(
+  auth: Authenticator,
+  {
+    title,
+    conversationId,
+    rows,
+  }: {
+    title: string;
+    conversationId: string;
+    rows: Array<CSVRecord>;
+  }
+): Promise<FileResource> {
+  const workspace = auth.getNonNullableWorkspace();
+  const user = auth.user();
+
+  let content: string = "";
+  const sections: Array<CoreAPIDataSourceDocumentSection> = [];
+
+  for (const row of rows) {
+    const rowId = typeof row["Id"] === "string" ? row["Id"] : "";
+    const rowName = typeof row["Name"] === "string" ? row["Name"] : "";
+    const rowContent = JSON.stringify(row);
+    const section: CoreAPIDataSourceDocumentSection = {
+      prefix: `${title} - Row ${rowId} ${rowName}`,
+      content: rowContent,
+      sections: [],
+    };
+    sections.push(section);
+    content += `${rowContent}\n`;
+  }
+
+  const file = await FileResource.makeNew({
+    workspaceId: workspace.id,
+    userId: user?.id ?? null,
+    contentType: "text/vnd.dust.attachment.searchable.text",
```

Comment:
searchable is not quite the right name here no?

It is a section format (at least that's the name we gave it so far) + I think we want a mimeType that is independent of being a attachment or not as we will want to reuse it in other context?

`text/vnd.dust.section-structured` maybe?